File: backend/src/main/java/com/example/jwt/backend/entites/ExaminationOffer.java
package com.example.jwt.backend.entites;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
@Entity
@Table
@Getter
@Setter
public class ExaminationOffer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column
    private String name;
    @Column(length = 100000)
    private String description;

    @Column(name = "price", precision = 10, scale = 2) // precision is the total number of digits, scale is the number of decimal places
    private BigDecimal price;
}

File: frontend/src/app/examinationOffer.ts
export class ExaminationOffer {
    constructor(
        public id: number | null,
        public name: string,
        public description: string,
        public price: number
    ) {}

    equals(other: ExaminationOffer): boolean {
        
        return this.id === other.id && this.name === other.name
      }
}